sdks/go/pkg/beam/core/runtime/exec/datasource.go

```diff
@@ -37,11 +37,13 @@ type DataSource struct {
 	Coder *coder.Coder
 	Out   Node
 
-	source   DataManager
-	state    StateReader
-	index    int64
-	splitIdx int64
-	start    time.Time
+	source DataManager
+	state  StateReader
+	// TODO(lostluck) 2020/02/06: refactor to support more general PCollection metrics on nodes.
+	outputPID string // The index is the output count for the PCollection.
```

Comment:
outputPID/PID isn't a great name for PCOllection ids since PID is used as PTrasnformId elsewhere.

Consider fixing here and elsewhere. Also, this comment should really say that this is the output pcollection id and not an implementation detail like this is the "index" since thats an implementation detail elsewhere about the mapping which this code shouldn't care about.

Reply:
Ack. Good point. 
This is to support the existing one off, and isn't going to be long term code. I'm already working on the stated TODO to handle metrics for all PCollections in the SDK. As it stands, the legacy metrics and monitoring infos use different identifiers for progress signals, so there's going to be a mess anyway.